trivia.py

```diff
@@ -461,17 +484,24 @@ def _get_new_question(self):
         '''
         damaged_question = True
```

Comment:
Generators are a kind of advanced thing. They behave like an iterator or a list object, but at their core are fundamentally not.

It's essentially a function-like object that retains its state, where control of execution is passed back and forth between it and the calling for-loop.

You use them as you would an iterable, so you say `for item in generator:` as they return an item to then be worked on in the for loop, but a generator is not indexable, so doing something like `generator[0]` won't work and will raise an exception.

Iteration in python happens through the `next()` method, which returns the next object to be worked on. In this case, we would create a generator that would simply choose and keep track of the next question for us, so that successive calls to the generator's `next()` method just returns the next question, without us having to keep track of anything in particular.

In this way, we can get rid of this while loop, and use a for loop to accomplish the same thing with fewer variables and it's usually safer.